h/static/scripts/group-forms/components/CreateGroupForm.tsx

```diff
@@ -64,20 +65,32 @@ function TextField({
   type: 'input' | 'textarea';
   value: string;
   onChangeValue: (newValue: string) => void;
-  limit: number;
+  minLength?: number;
+  maxLength: number;
   label: string;
   testid: string;
   required?: boolean;
   autofocus?: boolean;
   classes?: string;
 }) {
   const id = useId();
+  const [error, setError] = useState('');
```

Comment:
This works, although usually you want the state to be the [minimal state](https://react.dev/learn/thinking-in-react#step-3-find-the-minimal-but-complete-representation-of-ui-state) needed to determine the error condition, and then derive the actual message during rendering. As a contrived example, if the `minLength` changed after the commit, the error message should update to reflect that.

It occurred to me that here the state could probably be reduced to "has the user made an initial commit to this field"? The code below is not exactly the same logic, but does pass the tests and would probably be equally useful for users.

```diff
diff --git a/h/static/scripts/group-forms/components/CreateGroupForm.tsx b/h/static/scripts/group-forms/components/CreateGroupForm.tsx
index d85c7f2c5..7a01dbdf7 100644
--- a/h/static/scripts/group-forms/components/CreateGroupForm.tsx
+++ b/h/static/scripts/group-forms/components/CreateGroupForm.tsx
@@ -74,22 +74,23 @@ function TextField({
   classes?: string;
 }) {
   const id = useId();
-  const [error, setError] = useState('');
+
+  // True if an initial value has been committed.
+  const [hasCommitted, setCommitted] = useState(false);
 
   const handleInput = (e: InputEvent) => {
     onChangeValue((e.target as HTMLInputElement).value);
   };
 
   const handleChange = (e: Event) => {
-    if ((e.target as HTMLInputElement).value.length < minLength) {
-      setError(`Must be ${minLength} characters or more.`);
-    }
+    setCommitted(true);
   };
 
+  let error = '';
   if (value.length > maxLength) {
-    setError(`Must be ${maxLength} characters or less.`);
-  } else if (value.length >= minLength) {
-    setError('');
+    error = `Must be ${maxLength} characters or less.`;
+  } else if (value.length < minLength && hasCommitted) {
+    error = `Must be ${minLength} characters or more.`;
   }
 
   const InputComponent = type === 'input' ? Input : Textarea;
```

Reply:
That seems to work nicely, done